bindings/python/setup.py

```diff
@@ -46,8 +45,11 @@ def run(self):
             )
 
         cmake_version = re.search(r"version\s*([\d.]+)", out.decode().lower()).group(1)
-        if version.parse(cmake_version) < version.parse("3.10"):
-            raise RuntimeError("CMake >= 3.10 is required to build flashlight-text")
+        cmake_version_tuple = tuple([int(v) for v in cmake_version.split(".")])
+        if cmake_version_tuple < (3, 16):
+            raise RuntimeError(
+                f"CMake >= 3.16 is required to build flashlight-text; found {cmake_version}"
+            )
```

Comment:
Looks like that equivalent check is performed at the very beginning of cmake process. https://github.com/flashlight/text/blob/main/CMakeLists.txt#LL1C33-L1C33

Also it seems that two logic are out of sync before this commit.
Is there an advantage of manual check here?

Reply:
Fair point, and this is an annoying one -- biggest reason is just that the error handling from `pip install .` (and even `setup.py install`) is poor; the latter doesn't properly surface stderr in many cases, and both require a significant amount of scrolling to find the actual issue rather than a long trace

Reply:
oh I see. Then it makes sense to have the logic.